proposals/explicit-backing-fields.md

```diff
@@ -14,417 +14,396 @@ but one is part of a public API and another is an implementation detail.
+Primary property also might be `protected` or `internal`
+([example](https://github.com/meganz/android/blob/21a8dd8da80dfdf1dfed58ad1734a2c76fadac6f/app/src/main/java/mega/privacy/android/app/presentation/logout/LogoutViewModel.kt#L27-L28)).
 
-## Design
+### Delegation
 
-The proposed design consists of two new ideas: explicit backing fields and smart type narrowing.
+It's possible that either backing property ([example](https://github.com/b-lam/Resplash/blob/4b13d31134d1c31bd331e92ffe8d410984529212/app/src/main/java/com/b_lam/resplash/ui/upgrade/UpgradeViewModel.kt#L32-L40))
+or primary property ([example](https://github.com/google/accompanist/blob/0d0198d0cab599295f7601cb386d5989d72cc8cd/pager/src/main/java/com/google/accompanist/pager/PagerState.kt#L145))
+is delegated.
+
+_Statistics_: backing property delegation found in 2693 files, primary property delegation in 1010 files.
+
+## Design
 
 ### Explicit Backing Fields
 
-The grammar for `propertyDeclaration` is extended to support an optional _explicit backing field declaration_ in addition
-to the optional `getter` and `setter` (in any order between them). 
+By analogy with the already existing term "implicit backing fields", it is proposed to introduce a syntax
+for declaring "explicit backing fields" of properties.
+Syntax is consisted of a keyword `field`, optional type definition and initialization expression
+(more detailed [Grammar changes](#grammar-changes) are in the separate section).
 
+```kotlin
+private val _city = MutableLiveData<String>()
+val city: LiveData<String> get() = _city
 ```
-propertyDeclaration ::=
-    modifiers? ('val' | 'var') typeParameters?
-    (receiverType '.')?
-    (multiVariableDeclaration | variableDeclaration)
-    typeConstraints? (('=' expression) | propertyDelegate)? ';'?
-     ( (getter? (semi? setter)? (semi? field)?) 
-     | (setter? (semi? getter)? (semi? field)?)
-     | (getter? (semi? field)? (semi? setter)?) 
-     | (setter? (semi? field)? (semi? getter)?)
-     | (field? (semi? getter)? (semi? setter)?) 
-     | (field? (semi? setter)? (semi? getter)?)
 
-getter ::=
-    modifiers? 'get' ('(' ')' (':' type)? functionBody)?
-  
-setter ::=
-    modifiers? 'set' ('(' functionValueParameterWithOptionalType ','? ')' (':' type)? functionBody)?
-    
-field ::=
-    modifiers? 'field` (':' type)? ('=' expression)?       
+Could be rewritten as follows:
+
+```kotlin
+val city: LiveData<String>
+    field = MutableLiveData()
 ```
 
-Explicit backing field declaration has an optional visibility, an optional type, and an optional 
-initialization expression. 
+### Visibility
 
-#### Restrictions
+`private` is the default and the only allowed visibility for explicit backing fields
+(supporting other visibilities might be considered, see [Future enhancements](#future-enhancements)).
 
-There are the following additional semantic restrictions on the property declaration grammar:
+Visibility of property can be any, except `private`.
```

Comment:
Minor: let's say "must be more permissive than", because this statement would stay correct w.r.t. possible future enhancements